### ISSUE TITLE
Kernel#system fails with out: $stdout

### ISSUE THREAD

**Issue by elia** [jruby/jruby#5865]
### Environment

* Darwin Elia.local 18.7.0 Darwin Kernel Version 18.7.0: Tue Aug 20 16:57:14 PDT 2019; root:xnu-4903.271.2~2/RELEASE_X86_64 x86_64

### Expected Behavior

```
⤑ ruby -ve 'system("echo hello world", out: $stdout)'
ruby 2.6.2p47 (2019-03-13 revision 67232) [x86_64-darwin18]
hello world
⤑ 
```

### Actual Behavior

```
⤑ ruby -ve 'system("echo hello world", out: $stdout)'                          ~
jruby 9.2.6.0 (2.5.3) 2019-02-11 15ba00b Java HotSpot(TM) 64-Bit Server VM 25.25-b02 on 1.8.0_25-b17 +jit [darwin-x86_64]
NotImplementedError: cyclic redirects in child are not supported
  system at org/jruby/RubyKernel.java:1661
  <main> at -e:1
```

📝 Should be noted that this happens only using `$stdout`, with other `out:` destinations works fine:

```
⤑ ruby -ve 'system("echo hello world", out: $stderr)'                                                                                ~
jruby 9.2.6.0 (2.5.3) 2019-02-11 15ba00b Java HotSpot(TM) 64-Bit Server VM 25.25-b02 on 1.8.0_25-b17 +jit [darwin-x86_64]
hello world
```

```
⤑ ruby -ve 'system("echo hello world", out: File.new("/tmp/foo.txt","w"))'                                                           ~
jruby 9.2.6.0 (2.5.3) 2019-02-11 15ba00b Java HotSpot(TM) 64-Bit Server VM 25.25-b02 on 1.8.0_25-b17 +jit [darwin-x86_64]
⤑ cat /tmp/foo.txt                                                                                                                   ~
hello world
⤑                     
```

**Comment by headius** [comment]
Weird, I totally thought I commented on this but it didn't take.

So I think this can be fixed most likely...I was not clear how this "circular" redirection is supposed to work, so I think I mostly stubbed it out when I did the reimplementation of process launching.

**Comment by headius** [comment]
Yeah this is totally fixable. I disabled it because it was making modifications to file descriptors in the parent process which could potentially cause a race (primarily clearing O_CLOEXEC) but I think we can work around that since these are rare operations and mostly pretty benign.

**Comment by headius** [comment]
I'm reenabling the code in question even though it still makes CLOEXEC changes to descriptors in the parent process. We are already a little off when it comes to CLOEXEC setting, and even if this is a race (unset, fork, set) we're only talking about a race that would affect subprocess launching, and even then it can be reduced by serializing popen calls (typically *not* done heavily across threads).

**Comment by headius** [comment]
Going to call this fixed. It occurred to me that if people run into issues with concurrently twiddling CLOEXEC bit on a given fd, then maybe they shouldn't be concurrently trying to use that fd in subprocesses.

**Comment by headius** [comment]
@elia I have merged #5898 in for JRuby 9.2.9. I believe it will help some additional specs and MRI tests pass, but I have not updated tags/excludes yet.

**Comment by elia** [comment]
Thanks! That’s great!

No need to rush anyway, the workaround is simple enough 👍 

https://github.com/opal/opal/blob/c3f42f0168f9fd1bd31ba2cb1fbb8c0fa32dfca7/lib/opal/cli_runners/system_runner.rb
